### DISCUSSION TITLE
User with different Roles and Companies (Roles specific to company_id)

### DISCUSSION THREAD

**Discussion by muhammadtalhaishtiaq** [spatie/laravel-permission#2085] [Q&A]
Hi,
I need to implement a system where a super admin will create roles which will be generic for all companies
But now the companies can also create further roles with specific permissions, how can I do that

how to add company_id when a company is adding role and company_id is null when super admin is creating a role ??

Example:
System User -> Create Roles (For all Companies)
Lets take Company-A
Now company A can create further roles with its own ID(company_id) and those roles will only be shown to
Company A

I hope I explained well, It's possible with this package? But how ??

**Comment by erikn69** [comment]
Duplicate of #2072

**Comment by muhammadtalhaishtiaq** [comment]
> Duplicate of #2072

Haven't got any feasible and practical response on my this issue. That is why posted again.

**Comment by erikn69** [comment]
Just one hour is really a small time for an answer, don't be impatient, answer takes days sometimes
and read documentation or other discussions before open an issue

**Comment by systemsolutionweb** [comment]
This is not a bug
